From ccb19dc20f4bcf6e2c1727e3be1f32c90143503a Mon Sep 17 00:00:00 2001
From: Maher Khenissi <maherx.khenissi@intel.com>
Date: Mon, 12 Nov 2012 10:26:56 +0100
Subject: [PORT FROM MAIN] calibrator: fix klocwork issues

BZ: 67389

function 'strncpy' may incorrectly check buffer boundaries
and may overflow buffer 'ifreq.ifr_ifrn.ifrn_name' of
fixed size (16)

Change-Id: Ib8dc57861a8747e544a4e6749eec0877291293d3
Orig-Change-Id: Icbde790dd92f7f0ebd14ad746104c97154da9897
Signed-off-by: Maher Khenissi <maherx.khenissi@intel.com>
Reviewed-on: http://android.intel.com:8080/73555
Reviewed-by: Casasnovas, QuentinX <quentinx.casasnovas@intel.com>
Reviewed-by: lab_cactus <lab_cactus@intel.com>
Reviewed-by: De Cesco, Jonathan <jonathan.de.cesco@intel.com>
Reviewed-by: Trivelly, Jean <jean.trivelly@intel.com>
Tested-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
Reviewed-on: http://android.intel.com:8080/78005
Reviewed-by: Li, XiaojingX <xiaojingx.li@intel.com>
Tested-by: Li, XiaojingX <xiaojingx.li@intel.com>
---
 calibrator/plt.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/calibrator/plt.c b/calibrator/plt.c
index 0dc1da3..b430abe 100644
--- a/calibrator/plt.c
+++ b/calibrator/plt.c
@@ -113,6 +113,7 @@ static int isiffup(const char *vname)
 	if (skfd) {
 		struct ifreq ifr;
 		strncpy (ifr.ifr_name, vname, sizeof ifr.ifr_name);
+		ifr.ifr_name[sizeof(ifr.ifr_name)-1]='\0';
 		if (!ioctl (skfd, SIOCGIFFLAGS, &ifr)) {
 		   const short int flags = ifr.ifr_flags;
 		   if (flags & IFF_UP) ret=1;
@@ -130,6 +131,7 @@ static int setiffdown(const char *vname)
 	if (skfd) {
 		struct ifreq ifreq;
 		strncpy(ifreq.ifr_name, vname, sizeof(ifreq.ifr_name));
+		ifreq.ifr_name[sizeof(ifreq.ifr_name)-1]='\0';
 		if (!ioctl (skfd, SIOCGIFFLAGS, &ifreq)) {
 			short int flags = ifreq.ifr_flags;
 			flags &= ~IFF_UP;
-- 
1.7.4.1

